src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -72,37 +75,36 @@ pub fn compile_targets<'a, 'cfg: 'a>(targets: &[(&'a Target, &'a Profile)],
     try!(links::validate(deps));
 
     let dest = if build_config.release {"release"} else {"debug"};
-    let root = if resolve.root() == pkg.package_id() {
-        pkg
-    } else {
-        deps.iter().find(|p| p.package_id() == resolve.root()).unwrap()
-    };
+    let root = deps.iter().find(|p| p.package_id() == resolve.root()).unwrap();
```

Comment:
Hm the placement of the output directory should be relatively independent from the root package (e.g. it's just a default), as in once the output directory is chosen the root package is never needed after that. 

I'll try to audit for usage of this regardless, though, to see what's up.